sufia-models/app/models/concerns/sufia/collection_behavior.rb

```diff
@@ -15,10 +15,37 @@ def update_permissions
       self.visibility = "open"
     end
 
-    # Compute the sum of each file in the collection
-    # Return an integer of the result
+    # Compute the sum of each file in the collection using Solr to
+    # avoid having to hit Fedora 
+    #
+    # Returns an integer 
     def bytes
-      members.reduce(0) { |sum, gf| sum + gf.content.size.to_i }
+      query = ActiveFedora::SolrQueryBuilder.construct_query_for_rel(has_model: file_model)
+      args = {
+        fq: "{!join from=hasCollectionMember_ssim to=id}id:#{id}",
+        fl: "id, #{file_size_field}",
+        rows: members.count
+      }
+
+      files = ActiveFedora::SolrService.query(query, args)
+      files.reduce(0) { |sum, f| sum += f[file_size_field].to_i }
     end
+
+    protected
+      # Field to look up when locating the size of each file in Solr.
+      # Override for your own installation if using something different
+      def file_size_field
+        Solrizer.solr_name('file_size', stored_integer_descriptor)
+      end
+
+      # Override if you are storing your file size in a different way
+      def stored_integer_descriptor
+        Solrizer::Descriptor.new(:integer, :stored)
```

Comment:
This should point at `Sufia::GenericFileIndexingService::STORED_INTEGER` now, right?